docs/datagen/server/advancements.md

```diff
@@ -1,9 +1,43 @@
 Advancement Generation
 ======================
 
-[Advancements] can be generated for a mod by subclassing `AdvancementProvider` and overriding `#registerAdvancements`. Advancements can either be created and supplied manually or, for convenience, created using `Advancement$Builder`.
+[Advancements] can be generated for a mod by constructing a new `AdvancementProvider` and providing `AdvancementSubProvider`s. Advancements can either be created and supplied manually or, for convenience, created using `Advancement$Builder`. The provider must be [added][datagen] to the `DataGenerator`.
 
-After implementation, the provider must be [added][datagen] to the `DataGenerator`.
+```java
+// On the MOD event bus
+@SubscribeEvent
+public void gatherData(GatherDataEvent event) {
+    event.getGenerator().addProvider(
+        // Tell generator to run only when server data are generating
+        event.includeServer(),
+        output -> new MyAdvancementProvider(
+          output,
+          event.getLookupProvider(),
+          // Sub providers which generate the advancements
+          List.of(subProvider1, subProvider2, /*...*/),
+          event.getExistingFileHelper()
+        )
+    );
+}
+```
+
+`AdvancementSubProvider`
+------------------------
+
+An `AdvancementSubProvider` is responsible for generating advancements, containing a method which takes in a registry lookup and the writer (`Consumer<Advancement>`). A subtype of `AdvancementSubProvider` should take in an `ExistingFileHelper` to use while building an `Advancement`.
+
+```java
+// In some AdvancementSubProvider
```

Comment:
That is outdated, we've moved to using `ForgeAdvancementProvider` and `AdvancementGenerator` (which should be in the most recent commit of the docs). Additionally, it could be, but it would be incorrect to assume all types are of `MyAdvancementSubProvider` when they could be subtypes of `AdvancementSubProvider`.

Reply:
Whoops, I meant `In MyAdvancementSubProvider`, because the constructor calls for the class's name to be `MyAdvancementSubProvider`.